client-spark/spark-3/src/main/java/org/apache/spark/shuffle/celeborn/SparkShuffleManager.java

```diff
@@ -79,6 +91,36 @@ public class SparkShuffleManager implements ShuffleManager {
   private long sendBufferPoolCheckInterval;
   private long sendBufferPoolExpireTimeout;
 
+  // track appShuffleId -> shuffleId Set in executor for cleanup
+  static class AppShuffleIdTracker {
+    private ConcurrentHashMap<Integer, HashSet<Integer>> shuffleIdMap =
+        JavaUtils.newConcurrentHashMap();
+
+    public void track(int appShuffleId, int shuffleId) {
+      HashSet<Integer> shuffleIds =
+          shuffleIdMap.computeIfAbsent(
+              appShuffleId,
+              new Function<Integer, HashSet<Integer>>() {
```

Comment:
super nit: We can make it fairly succint with:
`shuffleIdMap.computeIfAbsent(appShuffleId, id -> new HashSet<>())` instead of `Function`